src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java

```diff
@@ -0,0 +1,104 @@
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.ligthandshadow.componentsystem.components.LASTeamComponent;
+import org.terasology.ligthandshadow.componentsystem.components.WinConditionCheckOnActivateComponent;
+import org.terasology.logic.characters.CharacterHeldItemComponent;
+import org.terasology.logic.common.ActivateEvent;
+import org.terasology.logic.inventory.InventoryManager;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.ControlWidget;
+import org.terasology.rendering.nui.NUIManager;
+import org.terasology.rendering.nui.databinding.ReadOnlyBinding;
+import org.terasology.rendering.nui.widgets.UILabel;
+import org.terasology.world.block.items.BlockItemComponent;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class ScoreSystem extends BaseComponentSystem {
+    private static final Logger logger = LoggerFactory.getLogger(ScoreSystem.class);
+
+    @In
+    private InventoryManager inventoryManager;
+
+    @In
+    private NUIManager nuiManager;
+
+    @In
+    private EntityManager entityManager;
+
+    private ControlWidget scoreScreen;
+    private EntityRef score;
+    private int redScore = 0;
+    private int blackScore = 0;
+    private UILabel blackScoreArea;
+    private UILabel redScoreArea;
+
+    @Override
+    public void postBegin() {
+        // Sets score screen bindings
+        scoreScreen = nuiManager.getHUD().getHUDElement("LightAndShadow:ScoreHud");
+        blackScoreArea = scoreScreen.find("blackScoreArea", UILabel.class);
+        blackScoreArea.bindText(new ReadOnlyBinding<String>() {
+            @Override
+            public String get() {
+                return String.valueOf(blackScore);
+            }
+        });
+        redScoreArea = scoreScreen.find("redScoreArea", UILabel.class);
+        redScoreArea.bindText(new ReadOnlyBinding<String>() {
+            @Override
+            public String get() {
+                return String.valueOf(redScore);
+            }
+        });
+    }
+
+    @Override
+    public void initialise() {
+        // Displays score UI on game start
+        nuiManager.getHUD().addHUDElement("ScoreHud");
+    }
+
+    @ReceiveEvent(components = {WinConditionCheckOnActivateComponent.class, LASTeamComponent.class})
+    public void onActivate(ActivateEvent event, EntityRef entity) {
+        LASTeamComponent baseTeamComponent = entity.getComponent(LASTeamComponent.class);
+        EntityRef player = event.getInstigator();
+        CharacterHeldItemComponent characterHeldItemComponent = player.getComponent(CharacterHeldItemComponent.class);
+        if (characterHeldItemComponent != null) {
+            EntityRef heldItem = characterHeldItemComponent.selectedItem;
+
+            //check to see if player has other team's flag
+            if (baseTeamComponent.team.equals(baseTeamComponent.RED) && heldItem.getComponent(BlockItemComponent.class).blockFamily.getURI().toString()
```

Comment:
Also, regarding the if : You have split it at a very wrong point. It legit took me 10 seconds to figure out that the second expression is continued on the second line - which is a lot.

Split the condition on &&, and convert `LightAndShadowResources:BlackFlag` and `LightAndShadowResources:redFlag` to their own constants. Make the second condition more readable if you can.

Reply:
Also, what is the need for `equalsIgnoreCase`? `equals` should be more than adequate for your purposes.

Here is the reason why: You've used `LightAndShadowResources:redFlag` at some point in this PR, make the entire system wonky.

There is a reason I insist on consistency across everything - to avoid things like this.